model_utils.py

```diff
@@ -65,22 +71,28 @@ def train(self, train_set=None, val_set=None):
         train_set = train_set if train_set is not None else self.train_set
         val_set = val_set if val_set is not None else self.val_set
 
+        if not self.class_balanced:
+            class_weights = {k: (self.batch_size / (self.num_types * v)) for k,v in self.abundances.items()}
+        else:
+            class_weights = {k: 1/self.num_types for k,v in self.abundances.items()}
+            
         print("starting to train")
         history = model.fit(
             train_set,
             epochs=self.num_epochs,
             validation_data=val_set,
-            verbose=1)
+            verbose=1,
+            class_weight=class_weights)
```

Comment:
can just do `class_weight=class_weights if not self.class_balanced else None` and remove the conditionals on lines 74-78